apps/tangle-dapp/src/pages/restake/delegate/StakeInput.tsx

```diff
@@ -55,13 +55,19 @@ const StakeInput: FC<Props> = ({
       return {};
     }
 
-    const amountRaw =
+    const amountBigInt =
       delegatorInfo.deposits[selectedAsset.id]?.amount ?? ZERO_BIG_INT;
 
-    const maxFormatted = +formatUnits(amountRaw, selectedAsset.decimals);
+    const delegatedBigInt =
+      delegatorInfo.delegations.find(
+        (delegation) => delegation.assetId === selectedAsset.id,
+      )?.amountBonded ?? ZERO_BIG_INT;
+
+    const availableBalance = amountBigInt - delegatedBigInt;
```

Comment:
The backend has a field called [`delegated_amount`](https://github.com/tangle-network/tangle/blob/59c719c14f6f3834546eb22c7b61b119dfd1989b/pallets/multi-asset-delegation/src/types/delegator.rs#L220-L221) that indicates the total delegated amount, which can be used to calculate the available balance. 

This issue has been addressed in PR https://github.com/tangle-network/dapp/pull/2788.